shared/src/business/useCases/pendingMotion/getPendingMotionDocketEntriesForCurrentJudgeInteractor.ts

```diff
@@ -0,0 +1,186 @@
+            docketEntry,
+            applicationContext,
+            currentDate,
+          ),
+      ),
+    );
+
+  const judgePendingMotions =
+    pendingMotionDocketEntriesOlderThan180DaysFromDynamo.filter(
+      removeMotionsThatHaveBeenHandledInDynamo,
+    );
+
+  const judgePendingMotionsWithWorksheet: FormattedPendingMotionWithWorksheet[] =
+    await attachDocketEntryWorkSheets(applicationContext, judgePendingMotions);
+
+  const uniquePendingMotions: FormattedPendingMotionWithWorksheet[] =
+    removeDuplicateDocketEntries(judgePendingMotionsWithWorksheet);
+
+  return {
+    docketEntries: uniquePendingMotions,
+  };
+};
+
+async function attachDocketEntryWorkSheets(
+  applicationContext: IApplicationContext,
+  docketEntries: FormattedPendingMotion[],
+): Promise<FormattedPendingMotionWithWorksheet[]> {
+  const docketEntryIds = docketEntries.map(
+    docketEntry => docketEntry.docketEntryId,
+  );
+
+  const docketEntryWorksheets = await applicationContext
+    .getPersistenceGateway()
+    .getDocketEntryWorksheetsByDocketEntryIds({
+      applicationContext,
+      docketEntryIds,
+    });
+
+  const docketEntryWorksheetDictionary = docketEntryWorksheets.reduce(
+    (accumulator, docketEntryWorksheet) => {
+      accumulator[docketEntryWorksheet.docketEntryId] = docketEntryWorksheet;
+      return accumulator;
+    },
+    {} as { [key: string]: RawDocketEntryWorksheet },
+  );
+
+  const docketEntriesWithWorksheets: FormattedPendingMotionWithWorksheet[] =
+    docketEntries.map(docketEntry => {
+      return {
+        ...docketEntry,
+        docketEntryWorksheet: docketEntryWorksheetDictionary[
+          docketEntry.docketEntryId
+        ] || { docketEntryId: docketEntry.docketEntryId },
+      };
+    });
+
+  return docketEntriesWithWorksheets;
+}
+
+function removeDuplicateDocketEntries(
+  docketEntries: FormattedPendingMotionWithWorksheet[],
+): FormattedPendingMotionWithWorksheet[] {
+  const [docketEntriesWithLead, soloDocketEntries] = partition(
+    docketEntries,
+    de => !!de.leadDocketNumber,
+  );
+
+  const uniqueDocketEntryDictionary = docketEntriesWithLead.reduce(
+    (accumulator, docketEntry) => {
+      if (!accumulator[docketEntry.docketEntryId]) {
+        accumulator[docketEntry.docketEntryId] = docketEntry;
+        return accumulator;
+      }
+      if (isLeadCase(docketEntry)) {
+        accumulator[docketEntry.docketEntryId] = docketEntry;
+        return accumulator;
+      }
+
+      return accumulator;
+    },
+    {} as { [key: string]: FormattedPendingMotionWithWorksheet },
+  );
+
+  return [...soloDocketEntries, ...Object.values(uniqueDocketEntryDictionary)];
+}
+
+function removeMotionsThatHaveBeenHandledInDynamo(
+  docketEntry: FormattedPendingMotion,
+) {
+  return (
+    docketEntry.pending &&
+    MOTION_EVENT_CODES.includes(docketEntry.eventCode) &&
+    docketEntry.daysSinceCreated >= 180
+  );
+}
+
+async function getLatestDataForPendingMotionsFromDynamo(
```

Comment:
I don't think we should be referencing Dynamo in the Interactors as we should not be bound to DynamoDB. That should be behind the persistence gateway. It may just be a renaming of this function that's needed